### DISCUSSION TITLE
ente cli - dbus session

### DISCUSSION THREAD

**Discussion by roughnecks** [ente-io/ente#1425] [Q&A] [answered]
Hello again :)

In order to use ente cli on a server we must run these commands:

```
eval "$(dbus-launch --sh-syntax)"

mkdir -p ~/.cache
mkdir -p ~/.local/share/keyrings # where the automatic keyring is created

# 1. Create the keyring manually with a dummy password in stdin
eval "$(printf '\n' | gnome-keyring-daemon --unlock)"

# 2. Start the daemon, using the password to unlock the just-created keyring:
eval "$(printf '\n' | /usr/bin/gnome-keyring-daemon --start)"
```

What I don't understand is how can I restart the daemon if I close the terminal buffer; if I just run the last command it doesn't work and if I run them all (without `mkdir` obviously), ente cli says this:

```
./ente admin list-users
Assuming `emailaddress` as the Admin
------------
Error: old admin token, please re-authenticate using `ente account add`
```

I can just do what it asks and then it starts working again, but I guess there must be a better way to handle it.

**Comment by ua741** [comment, marked as answer]
For certain admin actions, currently we always require a recent token. 
Other CLI commands, especially the one related to export and updating subscriptions should work fine.

**Reply by roughnecks** [reply]
I see, so it is not my fault :)
I hope there won't be any reason for using ente cli so often now that everything is set up.

👍 Thanks
